From 9737a14cec40c3304dbada8d9218e26e50a3d8c7 Mon Sep 17 00:00:00 2001
From: Evgenii Stepanov <eugenis@google.com>
Date: Wed, 30 Aug 2017 14:29:19 -0700
Subject: [PATCH 21/47] [cfi] Blacklist libvpx.

CFI does not recognize assembly functions as valid indirect call targets.

Test: no cfi checks in vp8_build_intra_predictors_mby_s
Bug: 64979440
Change-Id: I76797f688624f61e0adb6fb4597a4825f9c4c831
---
 Android.bp        | 1 +
 Android.bp.in     | 1 +
 cfi_blacklist.txt | 1 +
 3 files changed, 3 insertions(+)
 create mode 100644 cfi_blacklist.txt

diff --git a/Android.bp b/Android.bp
index df2e2b9..ad834eb 100644
--- a/Android.bp
+++ b/Android.bp
@@ -2151,6 +2151,7 @@ cc_library_static {
         diag: {
             cfi: true,
         },
+        blacklist: "cfi_blacklist.txt",
     },
 }
 
diff --git a/Android.bp.in b/Android.bp.in
index a2c70e0..ac6a464 100644
--- a/Android.bp.in
+++ b/Android.bp.in
@@ -93,6 +93,7 @@ cc_library_static {
         diag: {
             cfi: true,
         },
+        blacklist: "cfi_blacklist.txt",
     },
 }
 
diff --git a/cfi_blacklist.txt b/cfi_blacklist.txt
new file mode 100644
index 0000000..1393ae4
--- /dev/null
+++ b/cfi_blacklist.txt
@@ -0,0 +1 @@
+src:*external/libvpx/*
-- 
2.17.1

